### PR TITLE
Add install script for linux and update readme file

### PR DESCRIPTION
**Pull request by hos7ein**
For fast deploy Tile38 on Linux server I created install script.

- add install-linux.sh script
- Update README.md file

### CODE REVIEW

File: install-linux.sh [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+chown -R tile38:tile38  /run/tile38
+chmod -R 755 /run/tile38
+
+#create PID file
+touch /run/tile38/tile38.pid
+chown -R tile38:tile38  /run/tile38/tile38.pid
+chmod -R 664 /run/tile38/tile38.pid
+
+#create systemd file
+cat > /usr/lib/systemd/system/tile38.service << 'EOF'
```
Comment (stevelacy) [inline install-linux.sh]: 👎 Many systems still use init.d

Also `/user/lib/systemd` is reserved for distro-installed packages. Use `/etc/systemd/system` instead

File: install-linux.sh [hunk 2]
```diff
@@ -0,0 +1,49 @@
+#!/bin/bash
+
+#create tile38 user
+useradd -m -d /var/lib/tile38 -s /sbin/nologin -U tile38
```
Comment (stevelacy) [inline install-linux.sh]: This should be optional. I do not want a separate tile38 user on my dev machine.

File: README.md [hunk 3]
```diff
@@ -64,7 +64,19 @@ brew install tile38
 tile38-server
 ```
 
-### Building Tile38 
+### Linux
+
+For install Tile38 on Linux you can use `install-linux.sh` script and save your time!
```
Comment (stevelacy) [inline README.md]: "To install Tile39 on Linux"

File: install-linux.sh (hunk 2 above)
Comment (hos7ein) [reply install-linux.sh]: yeah you are right, but it's necessary for more security.

File: install-linux.sh (hunk 1 above)
Comment (hos7ein) [reply install-linux.sh]: yes. I have changed it.

File: README.md (hunk 3 above)
Comment (hos7ein) [reply README.md]: yes. I have changed it

**Comment by tidwall** [comment]
Sorry about the slow response to this PR. 

While I certainly like the idea of have install scripts, I'm conflicted on whether or not having it part of the main repository. I'm worried that this might give the impression that this is the best way (or may the only way) to install on linux. I'm not exclusively a linux user myself, so I'm not sure. 

Does this cover all flavors or linux? If we do have an install script, maybe it should encompass more systems than just linux?

**Comment by hos7ein** [comment]
Actually, This script (install-linux-systemd.sh) will work on all Linux-based systems which have systemd. although, in the future we can update it and or create new script for old Linux-based systems which have init.

**Comment by rshura** [comment]
The comments above are completely valid and mostly non-answered:
* The user creation should be optional:
** not all systems want to add a separate user for tile38, although it's not a bad idea;
** what if the user was already created when the previous version of tile38 was installed?
* Not all systems want to use systemd. Some still use upstart.  Some will want to use supervisord.

Plus I can add a few more points:
* Why install binaries under `/usr/bin` instead of `/usr/local/bin`?
* Where's the log going to go?
** I would add `/var/log/tile38-server.log` and make systemd redirect stdout and stderr to that file.
** Then you'd need log rotation set up.

All in all it feels like there's a lot of sys-admin decisions that can contribute to the exact installation scenario.  I'd either have a script that is all-flexible, or non at all. This could maybe be an example of a linux setup, but does not look like a good authoritative linux setup.

**Comment by tidwall** [comment]
@hos7ein I really appreciate the work you put into the PR, but after some consideration I've decided that this is outside the scope of the project. For now it's best to keep install scripts out of the main codebase as to avoid confusion around what is the authoritative process for installing.

**Pull request closed without merging**